akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -371,6 +371,23 @@ object Flow {
   def upcast[In, SuperOut, Out <: SuperOut, M](flow: Flow[In, Out, M]): Flow[In, SuperOut, M] =
     flow.asInstanceOf[Flow[In, SuperOut, M]]
 
+  /**
+   * Collect the value of [[Optional]] from the elements passing through this flow, empty [[Optional]] is filtered out.
+   *
+   * Adheres to the [[ActorAttributes.SupervisionStrategy]] attribute.
+   *
+   * '''Emits when''' the current [[Optional]]'s value is present.
+   *
+   * '''Backpressures when''' the value of the current [[Optional]] is present and downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   * * */
+  def flattenOptional[Out, In <: Optional[Out]](): Flow[In, Out, NotUsed] =
```

Comment:
I'm not sure this convenience is worth adding since it's not that difficult to compose from:
```
.filter(opt -> opt.isPresent())
.map(opt -> opt.get())
```

Reply:
![image](https://user-images.githubusercontent.com/501740/160777588-31a86f57-30ba-42f6-8874-4651dc5a77b7.png)
So this may not valid now ?

Reply:
I see, we discussed it already. Let's see what opinion @johanandren and others have nowadays 😄

Reply:
@hepin1989 did you check that the type bound actually works in Java? I have a vague memory that was the problem with an earlier try dealing with this.

Reply:
@johanandren Yes:
```java
  @Test
  public void notWorkingWithNoOptionalSource() {
    Source.range(1, 10)
        .via(Flow.flattenOptional())
        .runWith(Sink.seq(), system);
  }
```
![image](https://user-images.githubusercontent.com/501740/160901147-9ee906f1-2f44-4583-88ee-cfa911bb757f.png)


![image](https://user-images.githubusercontent.com/501740/160901371-86b6229c-e7be-47d7-ac0d-226497aaab36.png)

@johanandren I tried it locally, sorry for my windows setup.

Reply:
Good, thanks for double checking!